org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/detachable/DetachableParseResultBase.java

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Sebastian Zarnekow and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.nodemodel.detachable;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.nodemodel.ICompositeNode;
+import org.eclipse.xtext.nodemodel.ILeafNode;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.impl.CompositeNode;
+import org.eclipse.xtext.nodemodel.impl.CompositeNodeWithSemanticElement;
+import org.eclipse.xtext.nodemodel.impl.InternalNodeModelUtils;
+import org.eclipse.xtext.nodemodel.impl.SyntheticCompositeNode;
+import org.eclipse.xtext.parser.AbstractParseResult;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.Iterables;
+
+/**
+ * @since 2.35
+ */
+public abstract class DetachableParseResultBase<NMR extends NodeModelReference, Serialized> extends AbstractParseResult {
```

Comment:
no strong opinion but would "Abstract..." be better than "...Base" as a class name?

Reply:
Sure. Changed.